src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupElement.java

```diff
@@ -317,45 +317,69 @@ public int getOffsetInCurrentCigar() {
         return offsetInCurrentCigar;
     }
 
-    /** for some helper functions */
-    private enum Direction { PREV, NEXT }
+    /**  Direction of search for some helper functions */
+    @VisibleForTesting
+    enum Direction { PREV(-1), NEXT(1);
+        private final int increment;
+
+        public int getIncrement() {
+            return increment;
+        }
+
+        private Direction(final int increment){
+            this.increment = increment;
+        }
+    }
 
     /**
      * Helper function to get cigar elements between this and either the prev or next genomic position
      *
      * @param direction PREVIOUS if we want before, NEXT if we want after
      * @return a non-null list of cigar elements between this and the neighboring position in direction
      */
-    private LinkedList<CigarElement> getBetween(final Direction direction) {
-        final int increment = direction == Direction.NEXT ? 1 : -1;
-        LinkedList<CigarElement> elements = null;
+    private List<CigarElement> getBetween(final Direction direction) {
+        final int increment = direction.getIncrement();
+
+        final List<CigarElement> elements = new ArrayList<>();
         final List<CigarElement> cigarElements = read.getCigarElements();
         final int nCigarElements = cigarElements.size();
         for ( int i = currentCigarOffset + increment; i >= 0 && i < nCigarElements; i += increment) {
             final CigarElement elt = cigarElements.get(i);
             if ( ON_GENOME_OPERATORS.contains(elt.getOperator()) ) {
                 break;
             } else {
-                // optimization: don't allocate list if not necessary
-                if ( elements == null ) {
-                    elements = new LinkedList<>();
-                }
-
-                if ( increment > 0 ) {
-                    // to keep the list in the right order, if we are incrementing positively add to the end
-                    elements.add(elt);
-                } else {
-                    // counting down => add to front
-                    elements.addFirst(elt);
-                }
+                elements.add(elt);
             }
         }
-
-        // optimization: elements is null because nothing got added, just return the empty list
-        return elements == null ? new LinkedList<>() : elements;
+        if (increment < 0){
+            Collections.reverse(elements);
+        }
+        return elements;
     }
 
     /**
+     * Helper function to get cigar operator right next to this position but only for not-on-genome operators.
+     *
+     * @param direction PREVIOUS if we want before, NEXT if we want after
+     * @return the neighboring position in given direction or null if there is none
```

Comment:
`the next not-on-genome Cigar operator in the given direction, or null if there is none`